packages/server-wallet/e2e-testing/scripts/start-ganache.ts

```diff
@@ -83,14 +97,67 @@ async function setupGanache() {
       }
     });
   }, commandArguments.miningInterval);
-
   console.log(chalk.green(`Ganache started on port ${commandArguments.port}`));
 
   const endpoint = `http://localhost:${commandArguments.port}`;
 
   // Deploy the contracts and output to the artifact file
   const deployResults = await deploy(endpoint);
+
   await writeFile(commandArguments.artifactFile, deployResults);
 
+  setupUnhandledErrorListeners();
+  // ganache registers handlers for these events and intercepts them
+  // We want to avoid this so we can write out results before exiting so we remove all existing listeners
+  process.removeAllListeners('SIGINT');
+  process.removeAllListeners('SIGTERM');
+  process.removeAllListeners('beforeExit');
+
   console.log(chalk.green(`Contract artifacts written to ${commandArguments.artifactFile}`));
+
+  const roles = await getRoles(commandArguments.roleFile);
+  const destinations = Object.keys(roles).map(rId => roles[rId].destination);
+
+  // exitHook will run when the process gets terminated for whatever reason
+  exitHook(done =>
```

Comment:
Why did we need to add a new package for this? What's wrong with `process.on('exit', ...)`?

Reply:
`process.on('exit', ...)` is not `async` so we would have to introduce some mechanism to handle waiting for the async operation to complete. `async-exit-hook` does that for us.

Reply:
which async operation? writing to a file? You could avoid that with `jsonFile.writeFileSync`, right?

Reply:
We are also getting the on-chain balances (ie:  `provider.getBalance`)